plugins/modules/proxysql_query_rules_fast_routing.py

```diff
@@ -0,0 +1,470 @@
+            force_delete=dict(default=False, type='bool'),
+            save_to_disk=dict(default=True, type='bool'),
+            load_to_runtime=dict(default=True, type='bool')
+        ),
+        supports_check_mode=True
+    )
+
+    perform_checks(module)
+
+    login_user = module.params["login_user"]
+    login_password = module.params["login_password"]
+    config_file = module.params["config_file"]
+
+    cursor = None
+    try:
+        cursor, db_conn = mysql_connect(
+            module,
+            login_user,
+            login_password,
+            config_file,
+            cursor_class='DictCursor'
+        )
+    except mysql_driver.Error as e:
+        module.fail_json(
+            msg="unable to connect to ProxySQL Admin Module.. %s" % to_native(e)
+        )
+
+    proxysql_query_rule_fast_routing = ProxyQueryRuleFastRouting(module)
+    result = {}
+
+    result['state'] = proxysql_query_rule_fast_routing.state
+
+    if proxysql_query_rule_fast_routing.state == "present":
+        try:
+            if not proxysql_query_rule_fast_routing.check_rule_cfg_exists(cursor):
+                if proxysql_query_rule_fast_routing.config_data["username"] and \
+                   proxysql_query_rule_fast_routing.config_data["schemaname"] and \
+                   proxysql_query_rule_fast_routing.config_data["flagIN"] and \
+                   proxysql_query_rule_fast_routing.check_rule_pk_exists(cursor):
+                    proxysql_query_rule_fast_routing.update_rule(
+                        module.check_mode,
+                        result,
+                        cursor
+                    )
+                else:
+                    proxysql_query_rule_fast_routing.create_rule(
+                        module.check_mode,
+                        result,
+                        cursor
+                    )
+            else:
+                result['changed'] = False
+                result['msg'] = (
+                    "The rule already exists in "
+                    "mysql_query_rules_fast_routing "
+                    "and doesn't need to be updated."
+                )
+                result['rules'] = proxysql_query_rule_fast_routing.get_rule_config(cursor)
+
+        except mysql_driver.Error as e:
+            module.fail_json(
+                msg="unable to modify rule.. %s" % to_native(e)
+            )
+
+    elif proxysql_query_rule_fast_routing.state == "absent":
+        try:
+            existing_rules = proxysql_query_rule_fast_routing.check_rule_cfg_exists(cursor)
+            if existing_rules > 0:
+                if existing_rules == 1 or \
+                   proxysql_query_rule_fast_routing.force_delete:
+                    proxysql_query_rule_fast_routing.delete_rule(module.check_mode,
+                                                                 result,
+                                                                 cursor)
```

Comment:
```suggestion
                    proxysql_query_rule_fast_routing.delete_rule(module.check_mode, result, cursor)
```
We could make the object name a bit shorter because it's obvious that it's about proxysql, so `query_rule_fast_routing` or imo just `query_rule` would be (imo) good